.github/workflows/posix.yml

```diff
@@ -62,8 +59,27 @@ jobs:
             pandas_testing_mode: "deprecate"
             test_args: "-W error"
             name: "Numpy Dev"
+          # Test lowest & highest supported pyarrow
+          # Don't test pyarrow v2/3: Causes timeouts in read_csv engine
+          # even if tests are skipped/xfailed
+          - env_file: actions-38.yaml
+            pattern: "not single_cpu"
+            pyarrow_version: 5
+            name: "Min Pyarrow"
+          - env_file: actions-38.yaml
+            pattern: "single_cpu"
+            pyarrow_version: 5
+            name: "Min Pyarrow (single)"
+          - env_file: actions-38.yaml
+            pattern: "not single_cpu"
+            pyarrow_version: 7
+            name: "Latest Pyarrow"
```

Comment:
Since this eventually won't be the latest, similar better to name this "Pyarrow v7"